project/SparkBuild.scala

```diff
@@ -345,6 +349,60 @@ object Assembly {
   )
 }
 
+object PySparkAssembly {
+  import sbtassembly.Plugin._
+  import AssemblyKeys._
+
+  lazy val settings = Seq(
+    unmanagedJars in Compile += { BuildCommons.sparkHome / "python/lib/py4j-0.8.2.1-src.zip" },
```

Comment:
would be best to not hardcode this version here, but since we do that everywhere already (e.g. `bin/pyspark`, `PythonRunner`) this change by itself doesn't add much more technical debt.